shell/platform/android/io/flutter/view/FlutterMain.java

```diff
@@ -265,40 +272,48 @@ public void run() {
+    private static ApplicationInfo getApplicationInfo(@NonNull Context applicationContext) {
         try {
-            Bundle metadata = applicationContext.getPackageManager().getApplicationInfo(
-                applicationContext.getPackageName(), PackageManager.GET_META_DATA).metaData;
-            if (metadata != null) {
-                sAotSharedLibraryPath = metadata.getString(PUBLIC_AOT_AOT_SHARED_LIBRARY_PATH, DEFAULT_AOT_SHARED_LIBRARY_PATH);
-                sAotVmSnapshotData = metadata.getString(PUBLIC_AOT_VM_SNAPSHOT_DATA_KEY, DEFAULT_AOT_VM_SNAPSHOT_DATA);
-                sAotVmSnapshotInstr = metadata.getString(PUBLIC_AOT_VM_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_VM_SNAPSHOT_INSTR);
-                sAotIsolateSnapshotData = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_DATA_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_DATA);
-                sAotIsolateSnapshotInstr = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_INSTR);
-                sFlutterAssetsDir = metadata.getString(PUBLIC_FLUTTER_ASSETS_DIR_KEY, DEFAULT_FLUTTER_ASSETS_DIR);
-            }
+            return applicationContext
+                .getPackageManager()
+                .getApplicationInfo(applicationContext.getPackageName(), PackageManager.GET_META_DATA);
         } catch (PackageManager.NameNotFoundException e) {
             throw new RuntimeException(e);
         }
     }
 
-    private static void initResources(Context applicationContext) {
-        Context context = applicationContext;
-        new ResourceCleaner(context).start();
+    /**
+     * Initialize our Flutter config values by obtaining them from the
+     * manifest XML file, falling back to default values.
+     */
+    private static void initConfig(@NonNull Context applicationContext) {
+        Bundle metadata = getApplicationInfo(applicationContext).metaData;
 
-        Bundle metaData = null;
-        try {
-            metaData = context.getPackageManager().getApplicationInfo(
-                    context.getPackageName(), PackageManager.GET_META_DATA).metaData;
+        if (metadata == null) {
+            return;
+        }
 
-        } catch (PackageManager.NameNotFoundException e) {
-            Log.e(TAG, "Unable to read application info", e);
+        sAotSharedLibraryPath = metadata.getString(PUBLIC_AOT_AOT_SHARED_LIBRARY_PATH, DEFAULT_AOT_SHARED_LIBRARY_PATH);
+        sFlutterAssetsDir = metadata.getString(PUBLIC_FLUTTER_ASSETS_DIR_KEY, DEFAULT_FLUTTER_ASSETS_DIR);
+
+        sAotVmSnapshotData = metadata.getString(PUBLIC_AOT_VM_SNAPSHOT_DATA_KEY, DEFAULT_AOT_VM_SNAPSHOT_DATA);
+        sAotVmSnapshotInstr = metadata.getString(PUBLIC_AOT_VM_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_VM_SNAPSHOT_INSTR);
+        sAotIsolateSnapshotData = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_DATA_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_DATA);
+        sAotIsolateSnapshotInstr = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_INSTR);
```

Comment:
What do you think about returning a data structure that contains each of these fields, rather than assign a bunch of static fields? In general, I'd like to eliminate virtually all static stuff from this class, but that's a task for another day. In this particular case it seems like the purpose of this method is to lookup some info and then structure it. So having it return a new instance of a data structure, rather than mutate static state would probably be a good move in the right direction.

Reply:
Oh, and I'd probably take a 2nd look at the name `initConfig()` because, at least in my opinion, I probably wouldn't classify this method as initializing anything.

Reply:
I think this a good idea, what about we do that in a new PR? That might involve more changes that aren't related to this issue.

Reply:
Sure. You may want to file a bug to track it.